Automattic/newspack-plugin
# Stripe: send data about cancelled subscription to Campaigns

### All Submissions:






### Changes proposed in this Pull Request:



Adds "former donor" segmentation criterion handling, and updating Campaigns' segmentation data on cancelled subscription (recurring donation).

### How to test the changes in this Pull Request:

1. Start with a site with Stripe set as the Reader Revenue platform. The site should be reachable by Stripe webhooks
1. Switch `newspack-popups` to `feat/handle-subscription-cancellation` branch
2. Create a new segment, and configure it to target former donors in Reader Activity -> Donation section
3. Publish a prompt set to this segment 
4. On the frontend, in a new session, make a recurring donation
5. Via Stripe dashboard, cancel the subscription with immediate effect*
6. Load the site again, observe the prompt is displayed

\* self-cancellation via the billing portal won't work, because it only schedules a cancellation for the end of the current billing period.

### Other information:
